ethcore/types/src/transaction/transaction.rs

```diff
@@ -219,22 +354,11 @@ impl Transaction {
     pub fn with_signature(self, sig: Signature, chain_id: Option<u64>) -> UnverifiedTransaction {
```

Comment:
Reading EIP2718, it seems that typed transactions are designed to potentially be signatureless (_Another potential situation is one where transactions don’t have a signature in the traditional sense and instead have some other mechanism for determining validity._), so maybe signature methods `with_signature` or `null_sign`  belongs to `Transaction`, not to `TypedTransaction`

Reply:
Nonetheless, properly implementing this abstraction also breaks the current `SignedTransaction`/`UnverifiedTransaction`, just for something in the future that we do not know. So for me it's ok to leave as it is now.